File: server/data/results/translations/anchored/java/bubble_sort.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Main {

    public static List<Integer> bubble_sort(List<Integer> values) {
        List<Integer> items = new ArrayList<>(values);
        int n = items.size();
        for (int i = 0; i < n; i++) {
            boolean swapped = false;
            for (int j = 0; j < n - i - 1; j++) {
                if (items.get(j) > items.get(j + 1)) {
                    int temp = items.get(j);
                    items.set(j, items.get(j + 1));
                    items.set(j + 1, temp);
                    swapped = true;
                }
            }
            if (!swapped) {
                break;
            }
        }
        return items;
    }

    public static void main(String[] args) {
        List<Integer> data = Arrays.asList(5, 2, 9, 1, 5, 6);
        List<Integer> result = bubble_sort(data);
        String line = "";
        for (int value : result) {
            line = line + value + " ";
        }
        System.out.println(line.trim());
    }
}

File: server/data/results/translations/anchored/cpp/bubble_sort.cpp
#include <iostream>
#include <vector>
#include <string>

std::vector<int> bubble_sort(const std::vector<int>& values) {
    std::vector<int> items = values;
    int n = items.size();
    for (int i = 0; i < n; ++i) {
        bool swapped = false;
        for (int j = 0; j < n - i - 1; ++j) {
            if (items[j] > items[j + 1]) {
                int temp = items[j];
                items[j] = items[j + 1];
                items[j + 1] = temp;
                swapped = true;
            }
        }
        if (!swapped) {
            break;
        }
    }
    return items;
}

int main() {
    std::vector<int> data = {5, 2, 9, 1, 5, 6};
    std::vector<int> result = bubble_sort(data);
    std::string line = "";
    for (int value : result) {
        line = line + std::to_string(value) + " ";
    }
    size_t start = line.find_first_not_of(" \t\n\r");
    if (start != std::string::npos) {
        size_t end = line.find_last_not_of(" \t\n\r");
        line = line.substr(start, end - start + 1);
    } else {
        line = "";
    }
    std::cout << line << std::endl;
    return 0;
}